File: chapter-10/java/page-objects/src/test/java/com/manning/bddinaction/frequentflyer/acceptancetests/screenplay/registration/RegistrationForm.java
package com.manning.bddinaction.frequentflyer.acceptancetests.screenplay.registration;

import net.serenitybdd.screenplay.Question;
import net.serenitybdd.screenplay.abilities.BrowseTheWeb;
import net.serenitybdd.screenplay.questions.Text;
import net.serenitybdd.screenplay.targets.Target;

import java.util.Collection;

public class RegistrationForm {
    public static final Target EMAIL = Target.the("Email field").locatedBy("#email");
    public static final Target PASSWORD = Target.the("Password field").locatedBy("#password");
    public static final Target FIRSTNAME = Target.the("First name field").locatedBy("#firstName");
    public static final Target LASTNAME = Target.the("Last name field").locatedBy("#lastName");
    public static final Target ADDRESS = Target.the("Address field").locatedBy("#address");
    public static final Target COUNTRY = Target.the("Country field").locatedBy("#country");
    public static final Target TITLE = Target.the("Title dropdown").locatedBy("#title");
    public static final Target SEAT_PREFERENCE = Target.the("Seat preference").locatedBy("css:label[for='{0}']");
    public static final Target TERMS_AND_CONDITIONS = Target.the("Terms and conditions").locatedBy("#terms");
    public static final Target REGISTER = Target.the("Register button").locatedBy("//button[.='Register']");
    public static final Target FORM_ERROR_MESSAGES = Target.the("Error messages").locatedBy("css:mat-error");
    public static final Target ERROR_NOTIFICATION_MESSAGE = Target.the("Error message").locatedBy(".toast-error");

}

File: chapter-13/src/flying-high-app/registration/ui/registrationForm.ts
import { By, PageElement, PageElements } from '@serenity-js/web';

export const registrationForm = {
    emailField: () =>
        PageElement.located(By.id('email'))
            .describedAs('email field'),

    passwordField: () =>
        PageElement.located(By.id('password'))
            .describedAs('password field'),

    titleDropdown: () =>
        PageElement.located(By.id('title'))
            .describedAs('title dropdown'),

    firstNameField: () =>
        PageElement.located(By.id('firstName'))
            .describedAs('first name field'),

    lastNameField: () =>
        PageElement.located(By.id('lastName'))
            .describedAs('last name field'),

    addressField: () =>
        PageElement.located(By.id('address'))
            .describedAs('address field'),

    countryField: () =>
        PageElement.located(By.id('country'))
            .describedAs('country field'),

    seatPreferenceButtons: () =>
        PageElements.located(By.css(`[formcontrolname="seatPreference"]`))
            .describedAs(`seat preference radio buttons`),

    termsAndConditionsCheckbox: () =>
        PageElement.located(By.id('terms')),

    registerButton: () =>
        PageElement.located(By.id('register-button'))
            .describedAs('register button'),
}
